app/models/alchemy/essence_spree_variant.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Alchemy
+  class EssenceSpreeVariant < ActiveRecord::Base
+    VARIANT_ID = /\A\d+\z/
+
+    belongs_to :variant, class_name: 'Spree::Variant', optional: true
+
+    acts_as_essence(
+      ingredient_column: :variant,
+      preview_text_method: :name
+    )
+
+    def ingredient=(variant)
+      case variant
+      when VARIANT_ID
+        self.variant = Spree::Variant.new(id: variant)
```

Comment:
Should this be `.find_by(id: variant)`?

Reply:
We do not want to hit the DB while assigning, right? It is necessary to have an `Spree::Variant` instance so that we do not get an `ActiveRecord::AssociationTypeMismatch` (like tested in the specs) when assigning the variant.